services/datamanager/builtin/capture/file_count_logger.go

```diff
@@ -0,0 +1,79 @@
+package capture
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"time"
+
+	goutils "go.viam.com/utils"
+
+	"go.viam.com/rdk/data"
+	"go.viam.com/rdk/logging"
+	datasync "go.viam.com/rdk/services/datamanager/builtin/sync"
+)
+
+type fileCountLogger struct {
+	logger  logging.Logger
+	workers *goutils.StoppableWorkers
+}
+
+func newFileCountLogger(logger logging.Logger) *fileCountLogger {
+	return &fileCountLogger{
+		logger:  logger,
+		workers: goutils.NewBackgroundStoppableWorkers(),
+	}
+}
+
+func (poller *fileCountLogger) reconfigure(captureDir string) {
+	poller.workers.Stop()
+	poller.workers = goutils.NewBackgroundStoppableWorkers(func(ctx context.Context) {
```

Comment:
IIUC this creates exactly 1 worker (albeit inside a `StoppableWorkers` struct), maybe clearer to name the field `worker`?

Reply:
done